Kernel/Bus/PCI/Device.h

```diff
@@ -33,12 +52,23 @@ class Device {
 
     void enable_extended_message_signalled_interrupts();
     void disable_extended_message_signalled_interrupts();
+    ErrorOr<InterruptType> reserve_irqs(u8 number_of_irqs, bool msi);
+    ErrorOr<u8> allocate_irq(u8 index);
+    PCI::InterruptType get_interrupt_type();
+    void enable_interrupt(u8 irq);
+    void disable_interrupt(u8 irq);
 
 protected:
     explicit Device(DeviceIdentifier const& pci_identifier);
 
+private:
+    PhysicalAddress msix_table_entry_address(u8 irq);
+
 private:
     NonnullRefPtr<DeviceIdentifier const> const m_pci_identifier;
+    Interrupts m_interrupts;
+    bool m_msix_enabled { false };
+    bool m_msi_enabled { false };
```

Comment:
These two should be subsumed into a single `PCI::InterruptType`. As far as I can tell we only ever want one of them to be used at once.